maven-extension/src/main/java/io/opentelemetry/maven/resources/MavenResourceProvider.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.maven.resources;
+
+import io.opentelemetry.maven.semconv.MavenOtelSemanticAttributes;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+import org.apache.maven.rtinfo.RuntimeInformation;
+import org.apache.maven.rtinfo.internal.DefaultRuntimeInformation;
+
+public class MavenResourceProvider implements ResourceProvider {
+  @Override
+  public Resource createResource(ConfigProperties config) {
+    // TODO verify if there is solution to retrieve the RuntimeInformation instance loaded by the
```

Comment:
Hmm - the code is a bit simpler but the previous version seems to be more robust since it was using the real RuntimeInformation. It's one of the reasons for adding the resource customizer hook, for these situations where some class is used that's not available statically

Reply:
I'll iterate on this ASAP. I have to understand the best pattern to support default values provided by runtime components with  overwrites by the environment variables

I see an interesting impedance question to integrate the Java ServiceLoader SPI with dependency injection frameworks like `@inject`. Do you see things the same way?

Reply:
@cyrille-leclerc Yeah I don't think SPI works that well in the context of a system using dependency injection. So was hoping that the customizers on the builder will be able to solve those cases. Note that `service.name` is extremely special - it is one of few attributes automatically filled by the SDK, with the other ones having no use case for user customization.

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java#L48

So some ugly code handling it should be fine, the attribute itself is an ugly duckling, probably something like

```java
if (oldResource.get(SERVICE_NAME).equals(Resource.getDefault().get(SERVICE_NAME)) {
  newResource.put(SERVICE_NAME, MAVEN_SERVICE_NAME);
}
```

Long term if this is tricky in many cases, we could possibly consider adding a `serviceName` field to the SDK autoconfiguration builder that takes care of that logic, but for now we'd like to see how things go.

Reply:
Sorry @anuraaga I missed your comment, your point makes sense. Many thanks!